chat-widget/src/common/facades/FacadeChatSDK.ts

```diff
@@ -0,0 +1,331 @@
+                    Event: TelemetryEvent.NewTokenFailed,
+                    Description: ring.error?.message,
+                    ExceptionDetails: ring.error
+
+                });
+                return {
+                    result: false,
+                    message: ring.error?.message || "Failed to get token"
+                };
+            }
+
+        } catch (e: unknown) {
+
+            console.error("Unexpected error while getting token", e);
+
+            TelemetryHelper.logFacadeChatSDKEvent(LogLevel.ERROR, {
+                Event: TelemetryEvent.NewTokenFailed,
+                Description: "Unexpected error while getting token",
+                ExceptionDetails: e
+            });
+            return { result: false, message: "Unexpected error while getting token" };
+        }
+
+
+    }
+
+    private async validateAndExecuteCall<T>(functionName: string, fn: () => Promise<T>): Promise<T> {
+        const pingResponse = await this.tokenRing();
+
+        if (pingResponse.result === true) {
+            return fn();
+        }
+
+        //telemetry is already logged in tokenRing, so no need to log again, just return the error and communicate to the console
+        console.error("Authentication failed : Process to get a token failed for :" + functionName, pingResponse.message);
+
+        throw new Error("Authentication failed : Process to get a token failed for :" + functionName + " : " + pingResponse.message);
+    }
+
+    public async initialize(optionalParams: InitializeOptionalParams = {}): Promise<ChatConfig> {
+        return this.validateAndExecuteCall("initialize", () => this.chatSDK.initialize(optionalParams));
+    }
+
+    public async getChatReconnectContext(optionalParams: ChatReconnectOptionalParams = {}): Promise<ChatReconnectContext> {
+        return this.validateAndExecuteCall("getChatReconnectContext", () => this.chatSDK.getChatReconnectContext(optionalParams));
+    }
+
+    public async startChat(optionalParams: StartChatOptionalParams = {}): Promise<void> {
+        return this.validateAndExecuteCall("startChat", () => this.chatSDK.startChat(optionalParams));
+    }
+
+    public async endChat(optionalParams: EndChatOptionalParams = {}): Promise<void> {
+        return this.validateAndExecuteCall("endChat", () => this.chatSDK.endChat(optionalParams));
+    }
+
+    public async getCurrentLiveChatContext(): Promise<object> {
+        return this.validateAndExecuteCall("getCurrentLiveChatContext", () => this.chatSDK.getCurrentLiveChatContext());
+    }
+
+    public async getConversationDetails(optionalParams: GetConversationDetailsOptionalParams = {}): Promise<LiveWorkItemDetails> {
+        return this.validateAndExecuteCall("getConversationDetails", () => this.chatSDK.getConversationDetails(optionalParams));
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public async getPreChatSurvey(parse = true): Promise<any> {
```

Comment:
type this response

Reply:
ChatSDK returns any, facade propagates same response from origin